week2/homework/src/index.js

```diff
@@ -1,3 +1,58 @@
-'use strict';
-
-// TODO: Write the homework code in this file
+'use strict';
+const program = require('commander');
+const handler = require('./handler.js');
+
+var todos = handler.read();
+
+program
+    .version('0.1.0');
+
+program
+    .command('help')
+    .description('usage of this cmd line program')
+    .action(function (param) {
+        program.help();
+    });
+
+program
+    .command('add [todo]')
+    .description('add to-do')
+    .action(function (param) {
+        handler.add(param);
+        console.log('\"' + param + '\" saved');

```

Comment:
One problem I see here is that the `add` method is `async` (not as such but it calls `save` which calls the async `writeFile` method) and you're not waiting for its resolution to log `'\"' + param + '\" saved'`. So if there's an error, the user will first see the 'saved' message and then the error. 

You could make the method actually `async` and `await` it, or pass it a callback and do your `console.log` inside of it. I would use the second option, so that your handler methods only care about returning data or an error, and then the `action` functions would take care of logging both your success messages and your errors. 

Does that make sense?